src/screens/EventScreens/EventHomeScreen/index.js

```diff
@@ -107,18 +107,43 @@ const EventHomeScreen = ({ navigation }) => {
             noDataMessage="No Ongoing Events"
             eventData={filteredOngoingEvents}
             eventType={EVENT_TYPE.ONGOING}
+            onPress={(id) => {
```

Comment:
You should create a separate function and pass it to onPress as a prop to avoid unnecessary re-renders.
Also, can use the `find()` function for this purpose, which is more readable. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find